torch/distributed/optim/__init__.py

```diff
@@ -15,21 +16,12 @@
 from .functional_rmsprop import _FunctionalRMSprop
 from .functional_rprop import _FunctionalRprop
 from .functional_adamax import _FunctionalAdamax
+from .utils import (
+    functional_optim_map,
```

Comment:
Is there a particular reason why we treat `functional_optim_map` as a public variable? Is it for allowing users to extend it with other optim types?

Reply:
I think we should actually hide it, as if we wanted to support extension with other optim types there is a better approach for the developer (using the _OverlappedOptimizer interface built out in the next diff), and we're not planning on end users extending this map at least in the prototype version of this future.

To avoid issues with rebasing, I just went ahead and removed this in the top most diff.